packages/nimble-components/src/table/index.ts

```diff
@@ -210,6 +230,19 @@ export class Table<
         }
     }
 
+    public toggleGroupExpanded(rowIndex: number, event: Event): void {
+        const row = this.table.getRowModel().rows[rowIndex]!;
+        const wasExpanded = row.getIsExpanded();
+        // must update the collapsedRows before toggling expanded state
+        if (wasExpanded) {
+            this.collapsedRows.add(row.id);
+        } else {
+            this.collapsedRows.delete(row.id);
+        }
+        row.toggleExpanded();
+        event.stopPropagation();
```

Comment:
nit/pref:

I like that the toggleGroupExpanded method is a pretty self-contained except for this event reaction.

Maybe we could split them?
```ts
handleGroupRowExpanded (row, event) {
    this.toggleGroupExpanded(row);
    event.preventDefault();
}
```